src/dvc_objects/fs/generic.py

```diff
@@ -138,6 +138,28 @@ def _test_link(
     return True
 
 
+def _find_file(
+    from_fs: "FileSystem",
+    from_path: "",
+    size_limit: Optional[int] = None,
+) -> Optional["AnyFSPath"]:
+    try:
+        for root, _, files in from_fs.walk(from_path):
```

Comment:
So for every file transfered we will do all of these excessive calls. That will be bad perf wise. Seems like there should be a better way to solve it. The approach in this PR as is will not work out for us 🙂

Reply:
I'm sorry, the file size limit was intended for the "copy" mode, so that one doesn't copy GB's of data just to check links. The copy mode is nevertheless excluded anyways for now. So for the copy strategy now, no existing file would be used. What do you think is the best strategy here, is the copy mode even checked for usually?